### PR TITLE
MCMC densificastion and prunning strategy

### PR DESCRIPTION
**Pull request by zgojcic**

### CODE REVIEW

File: threedgrut/trainer.py [hunk 1]
```diff
@@ -411,9 +413,25 @@ def get_losses(
...
+        if self.conf.loss.use_scale:
+            with torch.cuda.nvtx.range(f"loss-scale"):
+                loss_scale = torch.abs(self.model.get_scale()).mean()
+                lambda_scale = self.conf.loss.lambda_scale
+
         # Total loss
-        loss = lambda_l1 * loss_l1 + lambda_l2 * loss_l2 + lambda_ssim * loss_ssim
-        return dict(total_loss=loss, l1_loss=loss_l1, l2_loss=loss_l2, ssim_loss=loss_ssim)
+        loss = lambda_l1 * loss_l1 + lambda_ssim * loss_ssim + lambda_opacity * loss_opacity + lambda_scale * loss_scale
+        return dict(total_loss=loss, l1_loss=lambda_l1 * loss_l1, l2_loss=lambda_l2 * loss_l2, ssim_loss=lambda_ssim * loss_ssim, opacity_loss=lambda_opacity * loss_opacity, scale_loss=lambda_scale * loss_scale)
```
Comment (zgojcic) [inline threedgrut/trainer.py:434]: This breaks comparisons on individual loss level to older commits, but I find it more informative to log the weights with their scaling factor applied

**Review by wilsonCernWq — APPROVED** [review]
I tweaked config and initialization strategy to stick to our old method. Perf is good on mipnerf360 and scannetpp for GUT. Some tests are still pending to run for 3DGRT, but overall it should be good to go.
